### ISSUE TITLE
Markdown Links are magled on repeated editing of /admin

### ISSUE THREAD

**Issue by igalic** [Plume-org/Plume#220]
After entering the following markdown into

```markdown
Esotericsystems is my non-technical, sometimes even creative out-let.

for technical musings look at [Brainsware](https://brainsware.at/blog/) or [Sealas](https://sealas.at/blog/)
```

trying to extend it later, it looked like so:

```markdown
Esotericsystems is my non-technical, sometimes even creative out-let.

for technical musings look at [Brainsware](https:&#x2F;&#x2F;brainsware.at&#x2F;blog&#x2F;) or [Sealas](https:&#x2F;&#x2F;sealas.at&#x2F;blog&#x2F;)
```

**Comment by BaptisteGelez** [comment]
I think it may be because `Instance.short_description` and `Instance.long_description` are `String`s and not `SafeString`s.

**Comment by igalic** [comment]
I'll try that out locally, and submit a patch if it works

**Comment by igalic** [comment]
```diff
diff --git a/plume-models/src/blogs.rs b/plume-models/src/blogs.rs
index a992f74..9c73a7d 100644
--- a/plume-models/src/blogs.rs
+++ b/plume-models/src/blogs.rs
@@ -22,6 +22,7 @@ use plume_common::activity_pub::{
     inbox::WithInbox,
     sign
 };
+use safe_string::SafeString;
 use instance::*;
 use users::User;
 use schema::blogs;
@@ -142,8 +143,8 @@ impl Blog {
                     name: inst.clone(),
                     local: false,
                     // We don't really care about all the following for remote instances
-                    long_description: String::new(),
-                    short_description: String::new(),
+                    long_description: SafeString::new(&<String>::new()),
+                    short_description: SafeString::new(&<String>::new()),
                     default_license: String::new(),
                     open_registrations: true,
                     short_description_html: String::new(),
diff --git a/plume-models/src/instance.rs b/plume-models/src/instance.rs
index 52830aa..a250306 100644
--- a/plume-models/src/instance.rs
+++ b/plume-models/src/instance.rs
@@ -3,6 +3,7 @@ use diesel::{self, QueryDsl, RunQueryDsl, ExpressionMethods, PgConnection};
 use std::iter::Iterator;
 
 use plume_common::utils::md_to_html;
+use safe_string::SafeString;
 use ap_url;
 use users::User;
 use schema::{instances, users};
@@ -16,8 +17,8 @@ pub struct Instance {
     pub blocked: bool,
     pub creation_date: NaiveDateTime,
     pub open_registrations: bool,
-    pub short_description: String,
-    pub long_description: String,
+    pub short_description: SafeString,
+    pub long_description: SafeString,
     pub default_license : String,
     pub long_description_html: String,
     pub short_description_html: String
@@ -30,8 +31,8 @@ pub struct NewInstance {
     pub name: String,
     pub local: bool,
     pub open_registrations: bool,
-    pub short_description: String,
-    pub long_description: String,
+    pub short_description: SafeString,
+    pub long_description: SafeString,
     pub default_license : String,
     pub long_description_html: String,
     pub short_description_html: String
diff --git a/plume-models/src/users.rs b/plume-models/src/users.rs
index 9cfad4e..9c2c811 100644
--- a/plume-models/src/users.rs
+++ b/plume-models/src/users.rs
@@ -205,8 +205,8 @@ impl User {
                     public_domain: inst.clone(),
                     local: false,
                     // We don't really care about all the following for remote instances
-                    long_description: String::new(),
-                    short_description: String::new(),
+                    long_description: SafeString::new(&<String>::new()),
+                    short_description: SafeString::new(&<String>::new()),
                     default_license: String::new(),
                     open_registrations: true,
                     short_description_html: String::new(),
diff --git a/src/routes/instance.rs b/src/routes/instance.rs
index 6905ecb..e96dfc0 100644
--- a/src/routes/instance.rs
+++ b/src/routes/instance.rs
@@ -3,6 +3,7 @@ use rocket::{request::LenientForm, response::Redirect};
 use rocket_contrib::{Json, Template};
 use serde_json;
 use validator::{Validate};
+use plume_models::safe_string::SafeString;
 
 use plume_models::{
     admin::Admin,
@@ -110,8 +111,8 @@ struct InstanceSettingsForm {
     #[validate(length(min = "1"))]
     name: String,
     open_registrations: bool,
-    short_description: String,
-    long_description: String,
+    short_description: SafeString,
+    long_description: SafeString,
     #[validate(length(min = "1"))]
     default_license: String
 }
diff --git a/src/setup.rs b/src/setup.rs
index 67a6d2f..8bc0eca 100644
--- a/src/setup.rs
+++ b/src/setup.rs
@@ -6,6 +6,7 @@ use std::io;
 use std::path::Path;
 use std::process::{exit, Command};
 use rpassword;
+use plume_models::safe_string::SafeString;
 
 use plume_models::{
     DB_URL,
@@ -152,8 +153,8 @@ fn quick_setup(conn: DbConn) {
         public_domain: domain,
         name: name,
         local: true,
-        long_description: String::new(),
-        short_description: String::new(),
+        long_description: SafeString::new(&<String>::new()),
+        short_description: SafeString::new(&<String>::new()),
         default_license: String::from("CC-0"),
         open_registrations: true,
         short_description_html: String::new(),
```

i'm not kinda stuck with:

```cargo
error[E0277]: the trait bound `plume_models::safe_string::SafeString: rocket::request::FromFormValue<'_>` is not satisfied               
   --> src/routes/instance.rs:109:10
    |
109 | #[derive(FromForm, Validate, Serialize)]
    |          ^^^^^^^^ the trait `rocket::request::FromFormValue<'_>` is not implemented for `plume_models::safe_string::SafeString`

error[E0308]: mismatched types
   --> src/routes/instance.rs:129:17
    |
129 |                 form.short_description.clone(),
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |                 |
    |                 expected struct `std::string::String`, found struct `plume_models::safe_string::SafeString`
    |                 help: try using a conversion method: `form.short_description.clone().to_string()`
    |
    = note: expected type `std::string::String`
               found type `plume_models::safe_string::SafeString`

error[E0308]: mismatched types
   --> src/routes/instance.rs:130:17
    |
130 |                 form.long_description.clone());
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |                 |
    |                 expected struct `std::string::String`, found struct `plume_models::safe_string::SafeString`
    |                 help: try using a conversion method: `form.long_description.clone().to_string()`
    |
    = note: expected type `std::string::String`
               found type `plume_models::safe_string::SafeString`

error: aborting due to 3 previous errors

Some errors occurred: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: Could not compile `plume`.                                                                                                        

To learn more, run the command again with --verbose.
```

and given that this is my first rust patch, i'm really not sure how to `impl` the `FromFormValue` for `SafeString` 😅  even tho it seems useful

**Comment by igalic** [comment]
fixed with #223
